packages/now-next/src/index.ts

```diff
@@ -1140,53 +1146,62 @@ export const build = async ({
 
       // redirects
       ...redirects,
-      // Before we handle static files we need to set proper caching headers
-      {
-        // This ensures we only match known emitted-by-Next.js files and not
-        // user-emitted files which may be missing a hash in their filename.
-        src: path.join(
-          '/',
-          entryDirectory,
-          '_next/static/(?:[^/]+/pages|chunks|runtime|css|media)/.+'
-        ),
-        // Next.js assets contain a hash or entropy in their filenames, so they
-        // are guaranteed to be unique and cacheable indefinitely.
-        headers: {
-          'cache-control': `public,max-age=${MAX_AGE_ONE_YEAR},immutable`,
-        },
-        continue: true,
-      },
-      { src: path.join('/', entryDirectory, '_next(?!/data(?:/|$))(?:/.*)?') },
 
       // Next.js page lambdas, `static/` folder, reserved assets, and `public/`
       // folder
       { handle: 'filesystem' },
 
-      // This needs to come directly after handle: filesystem to make sure to
-      // 404 and clear the cache header for _next requests
+      // We need to make sure to 404 for /_next after handle: miss since
+      // handle: miss is called before rewrites and to prevent rewriting /_next
+      { handle: 'miss' },
       {
         src: path.join(
           '/',
           entryDirectory,
           '_next/static/(?:[^/]+/pages|chunks|runtime|css|media)/.+'
         ),
-        headers: {
-          'cache-control': '',
-        },
         status: 404,
+        check: true,
+        dest: '$0',
       },
 
       ...rewrites,
```

Comment:
This still has to be after handle: filesystem

Reply:
It is, it's been moved up

Reply:
Check the diff again. It's after handle miss right now

Reply:
Are you seeing any outdated diff? `handle: 'filesystem'` comes directly after `redirects` in all cases.

https://github.com/zeit/now/blob/78a0a09c6703fd7501a334db5a52c24e2074f9f8/packages/now-next/src/index.ts#L1147-L1156

https://github.com/zeit/now/blob/78a0a09c6703fd7501a334db5a52c24e2074f9f8/packages/now-next/src/index.ts#L487-L497

Reply:
Oh I see, you mean it has to be before `handle: miss`

Reply:
When I say after handle: filesystem, I mean directly after

Reply:
Handle signals the start of a new phase

Reply:
Updated